core/app/models/spree/variant.rb

```diff
@@ -304,8 +304,13 @@ def should_track_inventory?
       self.track_inventory? && Spree::Config.track_inventory_levels
     end
 
+    # Image that can be used for the variant.
+    #
+    # Will first search for images on the variant, then all images from the
+    # products' variants. If all else fails, will return a new image object.
+    # @return [Spree::Image] the image to display
     def display_image
-      images.first || Spree::Image.new
+      images.first || product.variant_images.first || Spree::Image.new
```

Comment:
I see, I think I understand better now.

I think the problem here is the original intent of the deprecation, and why this method was introduced, was to move away from this behaviour. If you're editing stock for "white socks" it's confusing to have the "black socks" image shown. For this reason there was no fallback implemented in this method.

However, we probably often do want to use a fallback if there is no image directly attached to this variant, in the cart for example. I also think that when a fallback is desirable, if the product has an image associated (ie. an image on the master vairant) that should be preferred to the current fallback of "any image from any variant".

Perhaps being able to specify the callback would solve this issue. One possibility:

```
# For the cart, we want any image available
variant.display_image(fallbacks: [:product, :other_variant])

# Editing inventory, we want to make sure the image describes the exact variant
variant.display_image(fallbacks: [])
```

cc @richardnuno and @cbrunsdon who made this change originally.